onmt/modules/multi_headed_attn.py

```diff
@@ -246,45 +247,53 @@ def __init__(
         add_qkvbias=False,
         num_kv=0,
         use_ckpting=[],
+        parallel_gpu=1,
     ) -> None:
         assert model_dim % head_count == 0
         self.dim_per_head = model_dim // head_count
         super(MultiHeadedAttention, self).__init__()
         self.head_count = head_count
         self.num_kv = num_kv
+        self.parallel_gpu = parallel_gpu
         if num_kv == 0:
             self.linear_keys = skip_init(
                 nn.Linear,
                 in_features=model_dim,
-                out_features=model_dim,
+                out_features=model_dim // parallel_gpu,
```

Comment:
gentle reminder to put some assert somewhere for edge cases where model_dim might not be divisible by parallel_gpu (valid for other divisions below)

Reply:
will do.